library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -38,22 +39,30 @@
 
     private final ContentResolver resolver;
     private final DownloadDeleter downloadDeleter;
+    private final Uri batchContentUri;
+    private final Uri allDownloadsContentUri;
 
-    public BatchRepository(ContentResolver resolver, DownloadDeleter downloadDeleter) {
+    public static BatchRepository newInstance(ContentResolver resolver, DownloadDeleter downloadDeleter) {
```

Comment:
if this is conveniently hiding duplication then it is conveniently hiding a code smell

I think we should keep the constructor so it is big and ugly and obvious we need to refactor